tests/queries/0_stateless/02458_key_condition_not_like_prefix.sql

```diff
@@ -0,0 +1,5 @@
+CREATE TABLE data (str String) ENGINE=MergeTree ORDER BY str;
+INSERT INTO data (str) SELECT 'aa' FROM numbers(100000);
+INSERT INTO data (str) SELECT 'ba' FROM numbers(100000);
+INSERT INTO data (str) SELECT 'ca' FROM numbers(100000);
+SELECT count()FROM data WHERE str NOT LIKE 'a%' SETTINGS force_primary_key=1;
```

Comment:
`count() FROM`

Reply:
I think it would be nice to add some negative tests as well (i.e. test that an exception returned with force_primary_key = 1). Examples could include
- `NOT LIKE '%a'`
- `NOT LIKE 'a'`
- `NOT `LIKE '_a'`

etc.